Counterexamples/IrrationalPowerOfIrrational.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2024 Seewoo Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Seewoo Lee
+-/
+import Mathlib.Data.Real.Irrational
+import Mathlib.Analysis.SpecialFunctions.Pow.NNReal
+
+/-
+# Irrational power of irrational numbers are not necessarily irrational.
+
+Prove that there exist irrational numbers `a`, `b` such that `a^b` is rational.
+
+We use the following famous argument (based on the law of excluded middle and irrationality of √2).
+Consider `c = √2^√2`. If `c` is rational, we are done.
+If `c` is irrational, then `c^√2 = 2` is rational, so we are done.
+-/
+
+
+open Classical Real
+
+namespace Counterexample
+
+/--
+There exist irrational `a`, `b` with rational `a^b`.
+Note that the positivity assumption on `a` is required because of the definition of `rpow` for
+negative bases. See `Real.rpow_def_of_neg` for more details.
+-/
+theorem not_irrational_rpow : ¬ ∀ a b : ℝ, Irrational a → Irrational b → 0 < a → Irrational (a ^ b)
+    := by
+  intro h
+  by_cases hc : Irrational (√2 ^ √2)
+  · have ht := h (√2 ^ √2) √2 hc irrational_sqrt_two (NNReal.rpow_pos (by norm_num))
+    rw [← rpow_mul (by norm_num), mul_self_sqrt (by norm_num), rpow_two, sq_sqrt (by norm_num)]
+      at ht; simp at ht
+  · have ht := h √2 √2 irrational_sqrt_two irrational_sqrt_two (by norm_num)
+    exact hc ht
```

Comment:
To me, something like this feels a bit more idiomatic?
```suggestion
theorem not_irrational_rpow :
    ∃ a b : ℝ, 0 < a ∧ Irrational a ∧ Irrational b ∧ ¬ Irrational (a ^ b) := by
  by_cases hc : Irrational (√2 ^ √2)
  · use (√2 ^ √2), √2, by positivity, hc, irrational_sqrt_two
    rw [← rpow_mul, mul_self_sqrt, rpow_two, sq_sqrt] <;> norm_num
  · use √2, √2, by positivity, irrational_sqrt_two, irrational_sqrt_two, hc
```

Reply:
That's somewhat closer to what I initially did (which can be find in zulip - `∃` instead of `¬ ∀` - but also different and I like this better as it seems closer to the description). I simply followed @eric-wieser 's suggestion, so I wonder how you think about.

Reply:
The idea behind my suggestion is that starting with a `¬` is use elsewhere in the `Counterexamples` folder; the premise being that you're writing exactly the type that you would write for a regular theorem, but then putting a `not` before it to indicate it doesn't hold.

Reply:
In that case, you could still golf the proof slightly:
```suggestion
theorem not_irrational_rpow : ¬ ∀ a b : ℝ, Irrational a → Irrational b → 0 < a → Irrational (a ^ b)
    := by
  push_neg
  by_cases hc : Irrational (√2 ^ √2)
  · use (√2 ^ √2), √2, hc, irrational_sqrt_two, by positivity
    rw [← rpow_mul, mul_self_sqrt, rpow_two, sq_sqrt] <;> norm_num
  · use √2, √2, irrational_sqrt_two, irrational_sqrt_two, by positivity, hc
```

Reply:
Thanks! I didn't know about `push_neg` (and `positivity` also..)